_posts/2019-11-09-using-vault-with-strimzi.md

```diff
@@ -0,0 +1,184 @@
+
+## Separate intermediate CA per cluster
+
+Now that you have an intermediate CA, you might be tempted to use the same certificate and key for multiple clusters. However, considering Strimzi generates certificates outside of Vault, the only way you can revoke those certificates is by revoking the intermediate CA itself. As such, you should have _a separate intermediate CA per cluster._
+
+## Generating certificates using Vault and the intermediate CA
+
+If you want a client, for example, to be able to generate certificates using the intermediate CA, you'll need to set up a role for clients to use. More details on next steps [here](https://learn.hashicorp.com/vault/secrets-management/sm-pki-engine#step-3-create-a-role).
+
+# Launching your Kafka cluster using the Vault CA
+
+## Adding the certificate and private key to your Kubernetes cluster
+
+You need to create secrets _in your Strimzi namespace_ manually with the two files we generated in the previous few steps. Remember to include the labels, otherwise Strimzi will not pick them up.
+
+```bash
+# Private Key
+kubectl create secret -n ${strimziNamespace} generic ${clusterName}-cluster-ca \
+  --from-file=ca.key=intermediate.key.pem \
+  && kubectl label secret -n ${strimziNamespace} ${clusterName}-cluster-ca \
+  strimzi.io/kind="Kafka" \
+  strimzi.io/cluster="${clusterName}"
+
+# Certificate
+kubectl create secret -n ${strimziNamespace} generic ${clusterName}-cluster-ca-cert \
+  --from-file=ca.crt=intermediate.cert.pem \
+  && kubectl label secret -n ${strimziNamespace} ${clusterName}-cluster-ca-cert \
+  strimzi.io/kind="Kafka" \
+  strimzi.io/cluster="${clusterName}"
+```
+
+## Creating your Strimzi cluster
+
+You're almost there! Just a few more steps:
+
+1. Deploy the Strimzi operator
+2. Apply your `Kafka` CRD, remembering to set the following parameter:
+
+```yaml
+...
+spec:
+  clusterCa:
+    generateCertificateAuthority: false
+...
+```
+
+### A few things to note...
+
+* The `common_name` cannot contain a wildcard, so set it to something sensible. Otherwise you'll see something like this in your `tls-sidecar`:
+
+```text
+LOG5[1:139683950376704]: Service [zookeeper-2181] accepted connection from 127.0.0.1:33576
+LOG5[1:139683950376704]: connect_blocking: connected 10.100.241.50:2181
+LOG5[1:139683950376704]: Service [zookeeper-2181] connected remote server from 10.100.214.45:37700
+LOG4[1:139683950376704]: CERT: Verification error: permitted subtree violation
+LOG4[1:139683950376704]: Certificate check failed: depth=0, /O=io.strimzi/CN=zookeeper
+LOG3[1:139683950376704]: SSL_connect: 14090086: error:14090086:SSL routines:ssl3_get_server_certificate:certificate verify failed
+```
+
+* When generating your CA, `type` needs to be `exported`, otherwise the command will _not_ return the private key! [More information on that here.](https://www.vaultproject.io/api/secret/pki/index.html#generate-root)
```

Comment:
```suggestion
* When generating your intermediate CA, `type` needs to be `exported`, otherwise the command will _not_ return the private key! [More information on that here.](https://www.vaultproject.io/api/secret/pki/index.html#generate-root)
```